### DISCUSSION TITLE
Instruments to add to MuseSounds in 4.X

### DISCUSSION THREAD

**Discussion by DaffodilAura** [musescore/MuseScore#14022] [Ideas]
I'd like to see if we can get a set of standards for MuseSounds and let the community contribute, especially for folks looking for niche instruments- like me.  Among the obviously missing instruments for Muse Sounds are... 

- Alto Clarinet
- Bass Oboe
- Sarrusophones (SATBC)
- Piccolo Trumpet
- Bass Trumpet
- Soprano Trombone
- Alto Trombone
- Tenor Trombone
- Contrabass Trombone
- Euphonium
- Ocarina (SATBC)
- Panflute
- Pipe Organ (this instrument should have several sounds to it, including the Trompette, Hautbois, Basson and Vox Humana)
- Handbells

What other instruments would people like to see?  Let's collect ideas for Muse Sounds in this thread...  Do note that according to @MarcSabatella, the plan is to publish a document describing how to create Muse Sounds, and how to create software to use them, as well as to promote third party development.

**Comment by lgmventura** [comment]
- Lute
- Theorba
- Recorders (SATB)
- Natural horn
- Violas da Gamba 
- Baroque strings (solo and sections)

**Reply by DaffodilAura** [reply]
You mean this?
https://en.wikipedia.org/wiki/Theorbo

**Reply by topquark22** [reply]
I would like to have solo versions of the vocals (S, A, T, B, Boy Soprano solo), and in addition Girl Soprano. All in ensemble and solo voices.

**Comment by Fristover** [comment]
I know this is pretty hard to get under normal circumstances but any kind of bass oboe would be neat (20th century moment), early music instruments as mentioned above would be nice too, a solo contrabass is a must imo.

**Reply by DaffodilAura** [reply]
Added Bass Oboe to the list, since I can't get into the musescore.org forums...

**Comment by gerrybass** [comment]
i would like to add some techniques used for bass: funkstyle slap & pop, fingered and picked bass.

**Comment by Haru00007** [comment]
I often use the Hi B in Timpani (=Piccolo Timp.), but, MuseSounds can only play up to Hi A. So, I'd like you to add high sounds. 
And In instruments, I want drumset sounds (such as Hi-hat, Ride Cymbal)

**Comment by henkdegroot** [comment]
I would love to see:

Cornet (B-flat and E-flat)
Flugelhorn
Euphonium

**Reply by RyanK1729** [reply]
Euphonium would be excellent. There is pretty much the entirety of a standard concert band implemented with the exception of a Euphonium instrument.

**Comment by pmarx810** [comment]
Upright Bass for jazz scores

**Comment by pgrimes2002** [comment]
Euphonium for sure to help us who write for concert band and marching band. Also a mellophone sound. I know you can just use horn but they sound pretty different.

**Comment by pmarx810** [comment]
> What other instruments would people like to see?  Let's collect ideas for Muse Sounds in this thread...  Do note that according to @MarcSabatella, the plan is to publish a document describing how to create Muse Sounds, and how to create software to use them, as well as to promote third party development.

Just read this and think it's awesome. Having a page of community contributed instrument suggestions will be so cool and really speed up the implementation process.

**Comment by gracemaster** [comment]
I'd love to have a good "Hardanger Fiddle" and a "hurdy gurdy" and "nyckelharpa" They are great for Folk-inspired music.

**Comment by ghost** [comment]
I feel the biggest thing I'd want to see is Euphonium, it has a major part in concert bands, brass bands, and a handful of orchestral rep.

After that, flugelhorn and cornet would also be amazing to see, especially if we got Bb and Eb cornets for brass band.

Overall I'd like to see first and foremost that we can complete concert bands, as in my experience, they're one of the most common ensemble types to be written for.

**Comment by DaffodilAura** [comment]
I just realized that this is a feature request that the bot didn't pick up...

Also, here's a link for more suggestions...
https://musescore.org/en/node/336511

**Comment by sdsdsggg** [comment]
all the clarinet types (clarinet in A, picolo clarinet in D, contrabass clarinet etc.)
bass french horn
wagner tuba
also add musesounds for muted brass
musesound percussion:
guiro
canon
whip

**Reply by MicrotonalMatt** [reply]
Seconding muted brass. Even better if we could get a dedicated jazz trumpet sound with increased top range. The standard trumpet sound is sadly inadequate for big band writing.

**Comment by artemistic** [comment]
Euphonium, contrabass clarinet (please please please!), contralto clarinet, bass trumpet, and various mutes for brass (is stopped horn supported yet?)

**Comment by DaffodilAura** [comment]
Okay, I know it's been a while since I said anything on this issue, but I'd like to add a Lupophone to the list instruments available in MuseSounds.  Here's a video showing off the instrument...
https://www.youtube.com/watch?v=ihat57pZqP4

**Comment by Fourniture** [comment]
I would add: Finger snap, hand clap, and stomp for solo and group. It would be nice if these are also available through sound flags on every instrument.
Also:
waterphone
tuned anvil
tuned bottles
kalimba
at least three different sizes of cymbal and tam-tam
cork gun
air raid siren
nightingale whistle
Human whistle (quite necessary) 
Thunder sheet
Mouth harp
Wind machine
Steel pan
Taxi horns
Glass harp
(Yes, I would actually make use of all of these.)
Also Trumpet in C, which has a sound distinct from a b-flat trumpet.

**Reply by gogomuck** [reply]
I'd like see the alpine zither added to the list of instruments. I requested the addition over a year ago under feature request #16346. 

Alternatively, it would be great to have some mechanism within the program to add new instruments to the instruments list.

**Comment by andy-lyttle** [comment]
In addition to handbells, I'd like to see handchimes (not an instrument that currently exists in MuseScore Studio).

There are a ton of different handbell techniques that it would be nice to have support for, including:

- Shake
- Ring Touch
- Pluck
- Thumb Damp
- Mallet
- Mallet Roll
- Martellato
- Mart. Lift
- Echo
- Vibrato
- Swing
- Gyro
- Singing Bell

A few of these are also possible on handchimes, although the technique may be different (for example, Echo on handbells is achieved by touching the bell to a padded table, but on handchimes is achieved by touching the side of the chime with your finger).

**Comment by Rusca8** [comment]
Extremely niche, but here in catalonia it'd be useful to have sounds of [Cobla instruments](https://en.wikipedia.org/wiki/Cobla) for Sardana composers.

**Comment by Aurivox** [comment]
A clear, strong solo voice would be very nice, perhaps separated in soprano, alto, tenor, baritone and bass voice, in classical (more operatic) style and more pop/rock-alike style.

**Reply by DaddyLudwig** [reply]
I'd also love to have true solo voices to use in my scores that call for solo voices, such as those of Mozart's operas and Schubert's lieder. And of course, for Beethoven's Ninth which has that solo vs choir contrast. Right now, I'm having to use the choir sounds for both choir and solo, which is obviously not ideal.

**Comment by UsUsStudios** [comment]
I'm not sure if this is a MuseSounds thing or not, but I'd be interested in getting a few simple waveforms like sine, triangle etc. as instruments in MuseScore.

**Reply by MarcSabatella** [reply]
Those sounds are already in MS Basic.  It doesn't really make sense to have additional versions in MuseSounds since that library is so specifically about sampling of real players and capture those things that are *not* captured through ordinary synthesis techniques.

**Reply by UsUsStudios** [reply]
Oh, I understand now, I got that confused. Thank you for the clarification.

**Reply by nobody459** [reply]
I agree with UsUsStudios. Even if there's already sine/square/saw/etc. waves in MuseBasic they're... kinda bad ? For ex., there's a 0.3 second silence at the end of each sine wave note which sounds weird and unusual for an electronic instrument - you don't need 0.3 seconds to go from a F# to a G on a synth. It's instantaneous. And the others electronic instruments too are far from great.

It's not necessary to create a new musesounds just for that but improve the electronic instruments in MuseBasic would be both very easy for the developer and very handy for the users.

**Reply by MarcSabatella** [reply]
I'm not sure what you mean about a three second silence with the MS Basic sine wave - I just tried a simpel example and didn't hear anyuthing unusual.  But if you have a specific score that demonstrates a problem with the sine wave, be sure to open a new issue and ZIP and attach the score and describe the problem.  Same if you perceive problems in other sounds - but please make a separate issue for each.

**Reply by nobody459** [reply]
Thx for answering!
Not 3 seconds, 0.3 sec. (or so).
To test it, just add 2 untied whole notes to your score, you'll ear a very short silence between the 2.

I used that exemple because it's the only one i still use (since it's the best sounding of them all) but in my recollection, the other pure wave instruments also have their own issues - mostly sounding bad or not being faithful of what's expected of a square/sine/saw/etc. wave.

In addition, at least tens of instruments (mostly the electronic ones but not only) don't match their name and expected sound.
For ex. :
- the instrument called tam-tam (musebasic) is a sample of some other percussion instrument ;
- the teremin is litteraly a flute,
- etc.

I've personally given up on using all electronic instruments (except the sine) as they're all so bad compared to the strings section - even the musebasic one.

**Reply by MarcSabatella** [reply]
I hear nothing of the sort; each note last precisely as long as notated.  But I can guess what might be happening.  You may have added the instrument as "piano", and then changed the sound to sine wave in the Mixer, but it is still using the rest of the playback defaults oif piano, which includes a slight break between notes that don't have a slur.  The solution is simple - don't add a puinao instrument, but rather a synth instrument.

But, keep in mind MS Basic is based on the General MIDI standard, and indeed doesn't include sounds for most instruments not part of that standard.  That's why you don't hear a real tam-tam or theremind in that soundfont.  So, that's why MuseScore supports third party sounds.  But, any sound - electronic or otherwise - that *is* part of the General MIDI standard should work as expected.

Again, if you believe you have a score where this or any sound is not working correctly, please open an issue to report it so it can be fixed.  Be sure to ZIP and attach your score and be specific about what you perceive to be in error.

**Reply by nobody459** [reply]
> You may have added the instrument as "piano", and then changed the sound to sine wave in the Mixer, but it is still using the rest of the playback defaults oif piano, which includes a slight break between notes that don't have a slur.  

No i didn't.

> I hear nothing of the sort.

That's extremely surprising considering it's been a very stable 'bug' for me ever since i started using MS4 - unless it's intentional ? Maybe it depends on the configuration or something else idk, but there *is* a noticable silence between sine/square/etc. wave notes, and it's *not* due to adding that instrument as piano first and then changing it to a basic waveform.

> keep in mind MS Basic is based on the General MIDI standard, and indeed doesn't include sounds for most instruments not part of that standard.  That's why you don't hear a real tam-tam or theremind in that soundfont.

Yes i know. But
1. This 'standard' is not so standard considering it changes through time and between brands (e.g. Yamaha and Roland synths might not have the same number of melodic instruments, not maybe even the same instruments)
2. A standard is not a dictatorial rule is it ? 🙃

Just my thoughts...

**Reply by MarcSabatella** [reply]
As I said, if you are perceiving some sort of problem with the MS Basic sine wave sound in some particular score, please open up an issue, ZIP and attach your score, and then an investigation can begin.

General MIDI is very much a standard; it defines a set of sounds that all conforming devices and soundfonts provide.  I wasn’t merely explaining why the unavailable sounds are substituted.  It is true that sounds fonts can opt to provide additional sounds, so, again, if here is a specific sound you want to see added, please open a separate issue for each request.  Continuing to add comments in a discussion thread won’t accomplish anything.

**Comment by re0001** [comment]
Musical instruments from Latin America must be in Musescore 4.

Family : Wind
- Toyos 
- Quena
- Quenacho
- Quenilla
- Pifano
- Pifanillo
- Zampoña
- Zampoña (chromatic)
- Palla
- Malta
- Rondador

Family: Strings
- Charango
- Cuatro

Family: Percusion
-Bombo leguero
- Maracas
- Clave

Some of them require transposition.

**Comment by MarcSabatella** [comment]
Several of those instruments are already present, just not listed in the "common" section.  But you can find them if you switch to "all" instruments, or simply use the search bar.

For those *not* already present, if you'd like to see them added, please open an issue (of type "feature request") listing the specific instruments you would like along with all relevant info: the transposition, default clef, pitch range,etc.

**Reply by re0001** [reply]
Musical instruments from Latin America must be in Musescore 4.

This traditional instruments are used in the Orchestra of Andean Instruments in Ecuador.   Watch: https://www.youtube.com/watch?v=0lA2bOmKqfo

I send PDF file with information of the instruments.
[Latin America Instruments.pdf](https://github.com/user-attachments/files/23916174/Latin.America.Instruments.pdf)

**Comment by MarcSabatella** [comment]
As I explained, you need to open an issue too request this feature, and please include the necessary information directly.  Also your suggestion for which standard General MIDI sound be used for playback for each instrument.

**Reply by re0001** [reply]
Thanks for your support. I recently created the issue. Best regards.

**Comment by leedale30** [comment]
Would be nice to have a full GM sound set in musesounds to 'up grade' some of my old midi files from the 80s and 90s I have. Also maybe GM2, XG, GS, I forget the names now.

**Reply by DaffodilAura** [reply]
What is GM here?  I ask because I don't think I've heard of it before.  Also, sorry for the late response...

**Comment by MarcSabatella** [comment]
GM standards for General MIDI, the standard followed by most general purpose soundfonts (including MS Basic).  it defines a specific set of sounds and their numbering.

but note, MuseSounds issues are not tracked here, as it is not part of the open source software.  use their support site to make requests regarding that product.
